go/apps/mixnet/router.go

```diff
@@ -84,8 +94,54 @@ func (hp RouterContext) AcceptProxy() (net.Conn, error) {
 }
 
 // Close releases any resources held by the hosted program.
-func (hp RouterContext) Close() {
+func (hp *RouterContext) Close() {
 	if hp.proxyListener != nil {
 		hp.proxyListener.Close()
 	}
 }
+
+// HandleProxy reads a directive or a message from a proxy.
+func (hp *RouterContext) HandleProxy(c net.Conn) error {
+	var err error
+	cell := make([]byte, CellBytes)
+	if _, err = c.Read(cell); err != nil && err != io.EOF {
+		return err
+	}
+
+	// The first byte signals either a message or a directive to the router.
+	if cell[0] == msgCell {
+
+		msgBytes, n := binary.Uvarint(cell[1:])
```

Comment:
Given that clients are unauthenticated, this could be an attack, so we probably want to have a configurable bound on the maximum length of message we'll support.